rasa/utils/train_utils.py

```diff
@@ -619,6 +611,15 @@ def _check_confidence_setting(component_config: Dict[Text, Any]) -> None:
                 f"combination. You can use {MODEL_CONFIDENCE}={SOFTMAX} "
                 f"only with {SIMILARITY_TYPE}={INNER}."
             )
+    if component_config[MODEL_CONFIDENCE] == LINEAR_NORM:
+        rasa.shared.utils.io.raise_deprecation_warning(
+            f"{MODEL_CONFIDENCE} is set to `{LINEAR_NORM}`. This option was "
+            f"introduced in Rasa Open Source `2.3.0` but based on "
+            f"user feedback we have identified multiple problems with this "
+            f"option. Hence, `{MODEL_CONFIDENCE}={LINEAR_NORM}` is now deprecated "
+            f"and will be removed in Rasa Open Source 3.0.0. "
```

Comment:
Rephrasing: 
"We introduced this option in Rasa Open Source 2.3.0, but have identified multiple problems with it based on user feedback. `{MODEL_CONFIDENCE}={LINEAR_NORM}` is now deprecated and will be removed in Rasa Open Source 3.0.0."